src/game/spawn_object.c

```diff
@@ -124,8 +124,7 @@ struct Object *allocate_object(struct ObjectNode *objList) {
         // If no unimportant object exists, then the object pool is exhausted.
         if (unimportantObj == NULL) {
             // We've met with a terrible fate.
-            while (TRUE) {
-            }
+            ERROR("OBJECT POOL EXCEEDED");
```

Comment:
Does this _have_ to crash, or could it possibly get away with a debug assert and wager the risk of broken gameplay instead? The former seems worse than the gamble of the latter (especially when the debug assert exists for the developer)

Reply:
The `while (TRUE)` freezes the game anyways, it's not just broken gameplay. This just tells the user what happened.

Reply:
Vanilla's behavior makes this non-blocking, but at the same time that doesn't make a hard crash the objective best approach.

Reply:
True but IMO it's better than a freeze with no explanation.

Reply:
I agree but that's not the point I'm trying to make lol. I'm just comparing debug crash vs always crash.